compiler/dyno/lib/resolution/Resolver.cpp

```diff
@@ -662,6 +663,100 @@ QualifiedType Resolver::getTypeForDecl(const AstNode* declForErr,
   return QualifiedType(declKind, typePtr, paramPtr);
 }
 
+static bool isValidVarArgCount(QualifiedType paramSize) {
+  if (paramSize.type() == nullptr ||
+      paramSize.type()->isErroneousType()) {
+    return false;
+  } else if (paramSize.isParam() == false) {
+    return false;
+  } else {
+    const Param* p = paramSize.param();
+    if (p == nullptr) {
+      // param n : int, args...n
+      return true;
+    } else if (p->isIntParam()) {
+      int64_t val = p->toIntParam()->value();
+      return val > 0;
+    } else if (p->isUintParam()) {
+      uint64_t val = p->toUintParam()->value();
+      return val > 0;
+    } else {
+      return false;
+    }
+  }
```

Comment:
The problem was in ``varArgCountMatch`` in ``resolution-queries.cpp``. I just pushed a change that matches this behavior so that the function will fail to resolve. Still need a better error message, but that can wait for now.